mac/deploy_mac.sh

```diff
@@ -36,23 +36,40 @@ cleanup() {
 
 build_app() {
     local client_or_server="${1}"
-
-    # Build Jamulus
-    declare -a BUILD_ARGS=("_UNUSED_DUMMY=''")  # old bash fails otherwise
-    if [[ "${TARGET_ARCH:-}" ]]; then
-        BUILD_ARGS=("QMAKE_APPLE_DEVICE_ARCHS=${TARGET_ARCH}" "QT_ARCH=${TARGET_ARCH}")
-    fi
-    qmake "${project_path}" -o "${build_path}/Makefile" "CONFIG+=release" "${BUILD_ARGS[@]}" "${@:2}"
-    local target_name
-    target_name=$(sed -nE 's/^QMAKE_TARGET *= *(.*)$/\1/p' "${build_path}/Makefile")
     local job_count
     job_count=$(sysctl -n hw.ncpu)
 
-    make -f "${build_path}/Makefile" -C "${build_path}" -j "${job_count}"
+    # Build Jamulus for all requested architectures, defaulting to x86_64 if none provided:
+    local target_name
+    local target_arch
+    local target_archs_array
+    IFS=' ' read -ra target_archs_array <<< "${TARGET_ARCHS:-x86_64}"
+    for target_arch in "${target_archs_array[@]}"; do
+        qmake "${project_path}" -o "${build_path}/Makefile" \
+            "CONFIG+=release" \
+            "QMAKE_APPLE_DEVICE_ARCHS=${target_arch}" "QT_ARCH=${target_arch}" \
+            "${@:2}"
+        make -f "${build_path}/Makefile" -C "${build_path}" -j "${job_count}"
+        target_name=$(sed -nE 's/^QMAKE_TARGET *= *(.*)$/\1/p' "${build_path}/Makefile")
+        if [[ ${#target_archs_array[@]} -gt 1 ]]; then
+            # When building for multiple architectures, move the binary to a safe place to avoid overwriting by the other passes.
+            mv "${build_path}/${target_name}.app/Contents/MacOS/${target_name}" "${build_path}/${target_name}.app/Contents/MacOS/${target_name}.arch_${target_arch}"
+        fi
+        # We need to prune all object files here in order to force re-compilation in the next pass which builds
+        # for another architecture.
+        # We need to keep the ${build_path}/ contents though as we will use those artifacts later.
+        make -f "${build_path}/Makefile" -C "${build_path}" clean
```

Comment:
I'd do `make distclean` - it's more vicious.

Reply:
I'll try, I was worried about it being extra-correct and removing intermediate artifacts, but maybe it could work. I'll give it a try when I have time.

Reply:
Ok, as I feared: Simply running `make distclean` here fails, because it also removes the artifact which we want to keep:
https://github.com/jamulussoftware/jamulus/runs/8058626894?check_suite_focus=true#step:10:1464
`fatal error: /Applications/Xcode_13.4.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: can't open input file: /Users/runner/work/jamulus/jamulus/build/Jamulus.app/Contents/MacOS/Jamulus.arch_* (No such file or directory)`

Reply:
> If you need to keep anything, move it somewhere safe like on line 56.

I've now more or less implemented that -- I'm running `make distclean` right before the second (or later) pass' `qmake` (running it after `qmake` would possibly be incorrect, as `qmake` would have generated new cleanup instructions based on the next architecture).

This seems to work properly now:
https://github.com/jamulussoftware/jamulus/runs/8058721581?check_suite_focus=true#step:10:753

@pljones Are you fine with that?

Reply:
Yes, my standard build is
```
make distclean; qmake {optional params} && make j3 && make clean
```
except during a dev cycle, where I just do `make` a lot :).  We don't need the `make clean` as the container will get dropped.  And the way you've done it, the `make distclean` should always find a `Makefile` to work from.

Nice.